src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -131,42 +131,44 @@ bool MulticopterLandDetector::_get_ground_contact_state()
 		return true;
 	}
 
-	// land speed threshold
-	float land_speed_threshold = 0.9f * math::max(_params.landSpeed, 0.1f);
+	if (_vehicle_local_position.v_z_valid) {
```

Comment:
What happens if this is not valid?

Reply:
If the z velocity is not valid it will never show a `_state = LandDetectionState::GROUND_CONTACT;` state and the `vehicle_land_detected_s` uORB topic will never show a true ground contacted state.

In studying the code to answer your question, for the ground contact state to be true, `_vehicle_local_position.xy_valid` must also be true, (checked in the `_has_position_lock()`, and this is also being checked  in `_has_altitude_lock()` so I think this change can be reverted.  Thanks so much for pointing this out!  I'll push up a commit that reverts this change.